# Descarga estructura de la administración de la BDNS

Este cuaderno aspira a descargar la información relativa a la estructura de la administración que se ofrece desde el portal de la Base de Datos Nacional de Subvenciones. Para ello emplearemos la información recopilada en el siguiente issue: https://github.com/JaimeObregon/subvenciones/issues/15

De cara a generar unos ficheros reproducibles, vamos a generar dos opciones: trabajar con ficheros locales o descargarlos en caso de no estar disponibles, utilizando la librería de Python requests.

## Estructura de los datos
La base de datos nacional de subvenciones (BDNS) ofrece la información respecto a los convocantes de las subvenciones en tres niveles diferentes. Estos campos vienen recogidos en la base de datos en los campos:

- convocanteN1: que aparece etiquetado en el buscador como Administración
- convocanteN2: que aparece etiquetado en el buscador como Departamento
- convocanteN3: que aparece etiquedado en el buscador como Órgano

Se presupone una relación jerárquica a nivel administrativo entre los elementos en los niveles 1, 2 y 3.

## Información específica sobre convocantes de nivel 1
En la BDNS hay cuatro tipos de convocantes de nivel 1 diferentes:
- Administración del Estado (AE)
- Comunidades autónomas (CCAA)
- Entidades locales (EELL)
- Otros

### Administración del Estado
Esta compuesto por los distintos ministerios que forman parte del estado. Cada uno de estos ministerios está compuesto por una serie de órganos.

Ejemplo:
| ConvocanteN1 | ConvocanteN2                                                  | ConvocanteN3            |
|--------------|---------------------------------------------------------------|-------------------------|
| ESTADO       | MINISTERIO PARA LA TRANSICIÓN ECOLÓGICA Y EL RETO DEMOGRÁFICO | FUNDACIÓN BIODIVERSIDAD |

### Comunidades Autónomas
Esta compuesto por cada una de las 17 comunidades autónomas. Cada una de estas comunidades, está compuesta por una serie de departamentos.

| ConvocanteN1  | ConvocanteN2                 | ConvocanteN3  |
|---------------|------------------------------|---------------|
| ILLES BALEARS | INSTITUTO BALEAR DE JUVENTUD |               |


### Entidades locales
Está compuesto por los Ayuntamientos y otras entidades de caracter local. Tiene la particularidad de que al buscar por la Entidad Local, se presenta un nuevo cuadro en el que la entidad convocante, aparece adscrita a una entidad local, que suele coincidir con el nombre de la ciudad o el pueblo, y la provincia a nivel superior.

Ejemplo del resultado del buscador intermedio:

| Provincia | Entidad local | Convocante                    |
|-----------|---------------|-------------------------------|
|  ALAVA    | BERANTEVILLA  | AYUNTAMIENTO DE BERANTEVILLA  |

Ejemplo de entidad local en el buscador del BDNS:

| ConvocanteN1 | ConvocanteN2                 | ConvocanteN3 |
|--------------|------------------------------|--------------|
| BERANTEVILLA | AYUNTAMIENTO DE BERANTEVILLA |              |


### Otros
Está compuesto por todas las entidades que no están incluidas en los anteriores, tales como asociaciones, univerisdades, cámaras de comercio, etc.

Ejemplo:

| ConvocanteN1 | ConvocanteN2               | ConvocanteN3 |
|--------------|----------------------------|--------------|
|   OTROS      | CÁMARA DE COMERCIO DE LEÓN |              |


## Implicaciones respecto al tratamiento de datos
En el caso de los convocantesN1: Comunidades Autónomas, Entidades locales y Otros, no es posible seleccionar desde el buscador un tercer nivel de búsqueda. En estas búsquedas siempre encontraremos sólamente ConvocanteN1 y ConvocanteN2.

En el caso de que el convocante sea un ministerio, este aparecerá como ConvocanteN2. El convocanteN1 será siempre el "ESTADO" y el órgano convocante ocupará la columna de ConvocanteN3.

## Incoherencias en la interfaz y terminología de la BDNS
En la interfaz de búsqueda podemos encontrar las siguientes etiquetas para las columnas a las que nos referimos como "convocanteN1", "convocanteN2" y "convocanteN2": "Administración", "Departamento", "Organo". 

Esta denominación se emplea de forma inconsistente en el portal del BDNS. 

### Administración General del Estado:

En el cuadro de búsqueda aparece "A. del Estado" (Administración del Estado). Al marcar este cuadro, podemos elegir uno o más ministerios. Al hacerlo, se nos desbloquea el cuadro "Órgano".

En este caso encontramos la siguiente equivalencia:

| ConvocanteN1   | ConvocanteN2                                                  | ConvocanteN3            |
|----------------|---------------------------------------------------------------|-------------------------|
| Administración | Departamento                                                  | Órgano                  |
| ESTADO         | MINISTERIO PARA LA TRANSICIÓN ECOLÓGICA Y EL RETO DEMOGRÁFICO | FUNDACIÓN BIODIVERSIDAD |

Probablemente quedaría más clara esta equivalencia para la Administración General del Estado:

| ConvocanteN1   | ConvocanteN2                                                  | ConvocanteN3            |
|----------------|---------------------------------------------------------------|-------------------------|
| Administración | Ministerio                                                    | Órgano                  |
| ESTADO         | MINISTERIO PARA LA TRANSICIÓN ECOLÓGICA Y EL RETO DEMOGRÁFICO | FUNDACIÓN BIODIVERSIDAD |

### Comunidades Autónomas
En el cuadro de búsqueda aparece CC. Autónomas. Al seleccionar una o más Comunidades, se desbloquea el cuadro "Departamentos". En este caso, obtenemos los siguientes resultados:

| ConvocanteN1  | ConvocanteN2                 | ConvocanteN3  |
|---------------|------------------------------|---------------|
| Administración| Departamento                 | Órgano        |
| ILLES BALEARS | INSTITUTO BALEAR DE JUVENTUD |               |

En este caso, se puede observar que no es posible seleccionar el órgano, ya que esta casilla sólo está asociada a la AGE. Es importante tener en cuenta que en los resultados de la tabla no se indica que estamos tratando con una Comunidad Autónoma en ninguna parte (algo que si ocurre cuando seleccionamos como convocanteN1 la AGE).

### Entidades Locales
En el caso de las entidades locales, el encuadre es todavía más extraño.  Veamos la equivalencia con las etiquetas anteriores:

| ConvocanteN1   | ConvocanteN2                 | ConvocanteN3 |
|----------------|------------------------------|--------------|
| Administración | Departamento                 | Órgano       |
| BERANTEVILLA   | AYUNTAMIENTO DE BERANTEVILLA |              |

En este caso, el pueblo aparece como la Administración y el Ayuntamiento como un departamento de la misma. Al igual que en casos anteriores, no es posible elegir un órgano.

### Otros
En este caso, la estructura de nivel superior, es un identificador Otros que no identifica a ninguna Administración en concreto. La entidad que se selecciona aparece por defecto como convocanteN2.

### Conclusiones
En cada uno de los cuatro tipos de convocante, "Administración", "Departamento" y "Órgano" tienen un significado diferente, por lo que ser recomendable no emplear estos nombres y mantener la abstracción convocanteN1, N2 y N3 para denotar una dependecia jerárquica entre ambos (aunque esto no se cumple siquiera para el caso de OTROS)

# Relación entre los los niveles de convocante y la recuperación de datos
En la ayuda de la BDNS podemos encontrar instrucciones sobre [cómo generar hiperenlaces dinámicos al portal del Sistema Nacional de Publidad de Subevenciones y Ayudas Públicas, para empresas y organismos públicos.](https://www.infosubvenciones.es/bdnstrans/ayuda/pdf/Ayuda_Generacion_Hiperenlaces_SNPSAP)

Este documento describe cómo federar los datos disponibles de ayudas y subvenciones, de forma que se puedan generar contenidos dinámicos que apunten hacia el portal para sindicar contenidos concretos. A nivel práctico, esto permitiría que una entidad que está lanzando una convocatoria o una entidad beneficiaria, puedan ofrecer información actualizada sobre las subvenciones que están activas.

Uno de los ejemplos podrían ser [las subvenciones de las que resulta beneficiaria la Cruz Roja](https://www.infosubvenciones.es/bdnstrans/GE/es/concesiones/beneficiario/Q2866001G)

## Formato general de las consultas
Podemos encontrar toda la información sobre el tipo de consultas que podemos realizar, así como los códigos más habituales, en las páginas 2 y 3 del [documento](https://www.infosubvenciones.es/bdnstrans/ayuda/pdf/Ayuda_Generacion_Hiperenlaces_SNPSAP).

Generalizando esta información, podemos deducir que:
- Los códigos de la Administración General del Estado empiezan por C
- Los códigos de la Administración de las Comunidades Autónomas comienzan por A
- Los códigos para las Entidades Locales, están recogidos en un [excel independiente](https://www.oficinavirtual.pap.hacienda.gob.es/sitios/oficinavirtual/es-ES/CatalogoSistemasInformacion/TESEOnet/Documents/Códigos%20EELL%20hiperenlaces%20SNPSAP.xlsx) y comienzan por L.

Existe un identificador para C99 para Otros, pero no resulta obvio saber si se trata del mismo OTROS que encontramos en el buscador.

## TODO Ejemplos de consultas
Consulta de todas las convocatorias del Ministerio de Ciencia y Educación (en Español):

https://www.infosubvenciones.es/bdnstrans/GE/es/convocatorias/administracion/C31

## Consultas adicionales
Una vez disponemos de esta información, es posible aplicarla a otro formato de consultas del siguiente tipo:

https://www.infosubvenciones.es/bdnstrans/organos?code=C&value=31


Para obtener un listado de los organos dependientes del Ministerio en formato JSON.


In [2]:
import os
import pandas as pd

# Comprobamos si existe el fichero correspondiente en el directorio actual
# file_path = 'C31.csv'
#if os.path.exists(file_path) and os.stat(file_path).st_size == 0:

# Descargamos el fichero de organinos del Ministerio de Ciencia y Educación en un Dataframe:
# Nota: la URL vía https da fallo con pandas por el certificado

url_min_ciencia = 'https://www.infosubvenciones.es/bdnstrans/organos?code=C&value=31'

min_ciencia_df = pd.read_json(url_min_ciencia)

In [3]:
min_ciencia_df.head()

intidorgano strcodorgano  \
0         1967    EA0041268   
1         3319    EA0041187   
2         2041    EA0042469   
3         1770    EA0041170   
4         1768    EA0041267   

                                      strdescripcion stridadministracion  \
0  AGENCIA ESTATAL CONSEJO SUPERIOR DE INVESTIGAC...                   C   
1                   AGENCIA ESTATAL DE INVESTIGACIÓN                   C   
2  CENTRO DE INVESTIGACIONES ENERGÉTICAS, MEDIOAM...                   C   
3  CENTRO NACIONAL DE INVESTIGACIONES CARDIOVASCU...                   C   
4  CENTRO PARA EL DESARROLLO TECNOLÓGICO INDUSTRI...                   C   

   stragrupa  nivel1                              nivel2  \
0         31  ESTADO  MINISTERIO DE CIENCIA E INNOVACIÓN   
1         31  ESTADO  MINISTERIO DE CIENCIA E INNOVACIÓN   
2         31  ESTADO  MINISTERIO DE CIENCIA E INNOVACIÓN   
3         31  ESTADO  MINISTERIO DE CIENCIA E INNOVACIÓN   
4         31  ESTADO  MINISTERIO DE CIENCIA E INNOVACIÓN   

                                              nivel3  \
0  AGENCIA ESTATAL CONSEJO SUPERIOR DE INVESTIGAC...   
1                   AGENCIA ESTATAL DE INVESTIGACIÓN   
2  CENTRO DE INVESTIGACIONES ENERGÉTICAS, MEDIOAM...   
3  CENTRO NACIONAL DE INVESTIGACIONES CARDIOVASCU...   
4  CENTRO PARA EL DESARROLLO TECNOLÓGICO INDUSTRI...   

                     descripTrunc  \
0  AGENCIA ESTATAL CONSEJO SUP...   
1  AGENCIA ESTATAL DE INVESTIG...   
2  CENTRO DE INVESTIGACIONES E...   
3  CENTRO NACIONAL DE INVESTIG...   
4  CENTRO PARA EL DESARROLLO T...   

                                      descripTrunc65  
0  AGENCIA ESTATAL CONSEJO SUPERIOR DE INVESTIGAC...  
1                   AGENCIA ESTATAL DE INVESTIGACIÓN  
2  CENTRO DE INVESTIGACIONES ENERGÉTICAS, MEDIOAM...  
3  CENTRO NACIONAL DE INVESTIGACIONES CARDIOVASCU...  
4  CENTRO PARA EL DESARROLLO TECNOLÓGICO INDUSTRI...

In [4]:
min_ciencia_df.to_csv('C31-min-ciencia.csv')
min_ciencia_df.to_json('C31-min-ciencia.json')

Esta tabla nos muestra de forma ordenada mucha de la información que hemos visto más arriba:
- strcodorgano es el código Dir3
- stridadministracion nos indica que pertenecen a las administración general del estado
- stragrupa es el código del ministerio
- En nivel 3 y descripción encontramos descripciones más amplias
- En descripTrunc y descripTrunc65, versiones más cortas de la descripción

Nota para el Capitán Obvio: Obvio sabemos que C31 es este ministerio porque... nosotros mismos se lo hemos pasado en la URL! Pero si empezaramos a recuperar de forma secuencial, nos vendría bien para organizar esta información.

Probemos ahora con el código C99, que no sabíamos si coincidía con la campo de búsqueda otros.



In [5]:
url_otros = 'https://www.infosubvenciones.es/bdnstrans/organos?code=C&value=99'
otros_df = pd.read_json(url_otros)
otros_df.head()

intidorgano strcodorgano  \
0         4610    GA0000786   
1         4622    GA0000747   
2         4647    GA0000748   
3         4640    GA0000749   
4         4649    GA0000750   

                                      strdescripcion stridadministracion  \
0            TELENO - ASOCIACION MONTAÑAS DEL TELENO                   O   
1  ADATA - ASOCIACION PARA EL DESARROLLO DE ALIST...                   O   
2  ADECO BUREBA - ASOCIACION PARA EL DESARROLLO C...                   O   
3  ADECO CAMINO - ASOCIACION PARA EL DESARROLLO R...                   O   
4  ADECOAR - ASOCIACION PARA EL DESARROLLO DE LA ...                   O   

   stragrupa nivel1                                             nivel2  \
0         99  OTROS            TELENO - ASOCIACION MONTAÑAS DEL TELENO   
1         99  OTROS  ADATA - ASOCIACION PARA EL DESARROLLO DE ALIST...   
2         99  OTROS  ADECO BUREBA - ASOCIACION PARA EL DESARROLLO C...   
3         99  OTROS  ADECO CAMINO - ASOCIACION PARA EL DESARROLLO R...   
4         99  OTROS  ADECOAR - ASOCIACION PARA EL DESARROLLO DE LA ...   

   nivel3                    descripTrunc  \
0     NaN   TELENO - ASOCIACION MONTAÑ...   
1     NaN  ADATA - ASOCIACION PARA EL ...   
2     NaN  ADECO BUREBA - ASOCIACION P...   
3     NaN  ADECO CAMINO - ASOCIACION P...   
4     NaN  ADECOAR - ASOCIACION PARA E...   

                                      descripTrunc65  
0            TELENO - ASOCIACION MONTAÑAS DEL TELENO  
1  ADATA - ASOCIACION PARA EL DESARROLLO DE ALIST...  
2  ADECO BUREBA - ASOCIACION PARA EL DESARROLLO C...  
3  ADECO CAMINO - ASOCIACION PARA EL DESARROLLO R...  
4  ADECOAR - ASOCIACION PARA EL DESARROLLO DE LA ...

Viendo las primeras filas, !tiene buena pinta! Vamos a analizar mejor este dataframe.

In [6]:
otros_df.shape

(203, 10)

En total son 203 instituciones las clasificadas como Otras.

In [7]:
otros_df.head(-10)

intidorgano strcodorgano  \
0           4610    GA0000786   
1           4622    GA0000747   
2           4647    GA0000748   
3           4640    GA0000749   
4           4649    GA0000750   
..           ...          ...   
188         6462    U06300001   
189          521    U04000001   
190          540    U05500001   
191         6206    U05800001   
192         6414    U06400001   

                                        strdescripcion stridadministracion  \
0              TELENO - ASOCIACION MONTAÑAS DEL TELENO                   O   
1    ADATA - ASOCIACION PARA EL DESARROLLO DE ALIST...                   O   
2    ADECO BUREBA - ASOCIACION PARA EL DESARROLLO C...                   O   
3    ADECO CAMINO - ASOCIACION PARA EL DESARROLLO R...                   O   
4    ADECOAR - ASOCIACION PARA EL DESARROLLO DE LA ...                   O   
..                                                 ...                 ...   
188             UNIVERSIDAD INTERNACIONAL DE ANDALUCÍA                   O   
189                   UNIVERSIDAD JAUME I DE CASTELLÓN                   O   
190              UNIVERSIDAD MIGUEL HERNÁNDEZ DE ELCHE                   O   
191                       UNIVERSIDAD PABLO DE OLAVIDE                   O   
192               UNIVERSIDAD POLITÉCNICA DE CARTAGENA                   O   

     stragrupa nivel1                                             nivel2  \
0           99  OTROS            TELENO - ASOCIACION MONTAÑAS DEL TELENO   
1           99  OTROS  ADATA - ASOCIACION PARA EL DESARROLLO DE ALIST...   
2           99  OTROS  ADECO BUREBA - ASOCIACION PARA EL DESARROLLO C...   
3           99  OTROS  ADECO CAMINO - ASOCIACION PARA EL DESARROLLO R...   
4           99  OTROS  ADECOAR - ASOCIACION PARA EL DESARROLLO DE LA ...   
..         ...    ...                                                ...   
188         99  OTROS             UNIVERSIDAD INTERNACIONAL DE ANDALUCÍA   
189         99  OTROS                   UNIVERSIDAD JAUME I DE CASTELLÓN   
190         99  OTROS              UNIVERSIDAD MIGUEL HERNÁNDEZ DE ELCHE   
191         99  OTROS                       UNIVERSIDAD PABLO DE OLAVIDE   
192         99  OTROS               UNIVERSIDAD POLITÉCNICA DE CARTAGENA   

     nivel3                    descripTrunc  \
0       NaN   TELENO - ASOCIACION MONTAÑ...   
1       NaN  ADATA - ASOCIACION PARA EL ...   
2       NaN  ADECO BUREBA - ASOCIACION P...   
3       NaN  ADECO CAMINO - ASOCIACION P...   
4       NaN  ADECOAR - ASOCIACION PARA E...   
..      ...                             ...   
188     NaN  UNIVERSIDAD INTERNACIONAL D...   
189     NaN  UNIVERSIDAD JAUME I DE CAST...   
190     NaN  UNIVERSIDAD MIGUEL HERNÁNDE...   
191     NaN  UNIVERSIDAD PABLO DE OLAVID...   
192     NaN  UNIVERSIDAD POLITÉCNICA DE ...   

                                        descripTrunc65  
0              TELENO - ASOCIACION MONTAÑAS DEL TELENO  
1    ADATA - ASOCIACION PARA EL DESARROLLO DE ALIST...  
2    ADECO BUREBA - ASOCIACION PARA EL DESARROLLO C...  
3    ADECO CAMINO - ASOCIACION PARA EL DESARROLLO R...  
4    ADECOAR - ASOCIACION PARA EL DESARROLLO DE LA ...  
..                                                 ...  
188             UNIVERSIDAD INTERNACIONAL DE ANDALUCÍA  
189                   UNIVERSIDAD JAUME I DE CASTELLÓN  
190              UNIVERSIDAD MIGUEL HERNÁNDEZ DE ELCHE  
191                       UNIVERSIDAD PABLO DE OLAVIDE  
192               UNIVERSIDAD POLITÉCNICA DE CARTAGENA  

[193 rows x 10 columns]

Vemos que en este caso, el primer dígito se inspira en el CIF, aunque no coincide. (G indentifica asociaciones, pero U en el CIF identifica Uniones Temporales de Empresas, no Universidades).

In [8]:
otros_df['strcodorgano']

0      GA0000786
1      GA0000747
2      GA0000748
3      GA0000749
4      GA0000750
         ...    
198    U03700001
199    U01800001
200    U02700001
201    GA0000789
202    GA0000790
Name: strcodorgano, Length: 203, dtype: object

Antes de modificar los datos originales, vamos a guardar esta información localmente

In [9]:
otros_df.to_csv('C99-otros.csv')
otros_df.to_json('C99-otros.json')

Extraigo los dos primeros caracteres en una nueva columna:

In [10]:
otros_df['tipo_organo'] = otros_df['strcodorgano'].str[:2]

In [11]:
otros_df['tipo_organo'].astype('category')
otros_df['tipo_organo'].value_counts()

CC    71
U0    51
GA    49
I0    29
A1     2
GE     1
Name: tipo_organo, dtype: int64

Vamos a explorar las distintas categorias manualmente:

In [12]:
otros_df[otros_df['tipo_organo'] == 'CC'].head(5)

intidorgano strcodorgano  \
39         5578    CC0000008   
40         5556    CC0000004   
60         5747    CC0000012   
61         5303    CC0000013   
62         5292    CC0000014   

                                       strdescripcion stridadministracion  \
39  CONSEJO REGIONAL DE CÁMARAS DE COMERCIO DE AND...                   O   
40  CONSEJO REGIONAL DE CÁMARAS DE COMERCIO DE CAS...                   O   
60                        CÁMARA DE COMERCIO DE ALCOY                   O   
61                     CÁMARA DE COMERCIO DE ALICANTE                   O   
62                      CÁMARA DE COMERCIO DE ALMERÍA                   O   

    stragrupa nivel1                                             nivel2  \
39         99  OTROS  CONSEJO REGIONAL DE CÁMARAS DE COMERCIO DE AND...   
40         99  OTROS  CONSEJO REGIONAL DE CÁMARAS DE COMERCIO DE CAS...   
60         99  OTROS                        CÁMARA DE COMERCIO DE ALCOY   
61         99  OTROS                     CÁMARA DE COMERCIO DE ALICANTE   
62         99  OTROS                      CÁMARA DE COMERCIO DE ALMERÍA   

    nivel3                    descripTrunc  \
39     NaN  CONSEJO REGIONAL DE CÁMARAS...   
40     NaN  CONSEJO REGIONAL DE CÁMARAS...   
60     NaN     CÁMARA DE COMERCIO DE ALCOY   
61     NaN  CÁMARA DE COMERCIO DE ALICA...   
62     NaN  CÁMARA DE COMERCIO DE ALMER...   

                                       descripTrunc65 tipo_organo  
39  CONSEJO REGIONAL DE CÁMARAS DE COMERCIO DE AND...          CC  
40  CONSEJO REGIONAL DE CÁMARAS DE COMERCIO DE CAS...          CC  
60                        CÁMARA DE COMERCIO DE ALCOY          CC  
61                     CÁMARA DE COMERCIO DE ALICANTE          CC  
62                      CÁMARA DE COMERCIO DE ALMERÍA          CC

In [13]:
otros_df[otros_df['tipo_organo'] == 'I0'].head(5)

intidorgano strcodorgano                           strdescripcion  \
19         5533    I00001980  AGENCIA DE DESARROLLO DE LEA-ARTIBAI SA   
20         2814    I00000121  AGENCIA ESPAÑOLA DE PROTECCIÓN DE DATOS   
27         5951    I00001168         AUDIENCIA DE CUENTAS DE CANARIAS   
32         2920    I00000529                 CENTRE D'ESTUDIS PORCINS   
36          139    I00000177        CONSEJO CONSULTIVO DE EXTREMADURA   

   stridadministracion  stragrupa nivel1  \
19                   O         99  OTROS   
20                   O         99  OTROS   
27                   O         99  OTROS   
32                   O         99  OTROS   
36                   O         99  OTROS   

                                     nivel2  nivel3  \
19  AGENCIA DE DESARROLLO DE LEA-ARTIBAI SA     NaN   
20  AGENCIA ESPAÑOLA DE PROTECCIÓN DE DATOS     NaN   
27         AUDIENCIA DE CUENTAS DE CANARIAS     NaN   
32                 CENTRE D'ESTUDIS PORCINS     NaN   
36        CONSEJO CONSULTIVO DE EXTREMADURA     NaN   

                      descripTrunc                           descripTrunc65  \
19  AGENCIA DE DESARROLLO DE LE...  AGENCIA DE DESARROLLO DE LEA-ARTIBAI SA   
20  AGENCIA ESPAÑOLA DE PROTECC...  AGENCIA ESPAÑOLA DE PROTECCIÓN DE DATOS   
27  AUDIENCIA DE CUENTAS DE CAN...         AUDIENCIA DE CUENTAS DE CANARIAS   
32        CENTRE D'ESTUDIS PORCINS                 CENTRE D'ESTUDIS PORCINS   
36  CONSEJO CONSULTIVO DE EXTRE...        CONSEJO CONSULTIVO DE EXTREMADURA   

   tipo_organo  
19          I0  
20          I0  
27          I0  
32          I0  
36          I0

In [14]:
otros_df[otros_df['tipo_organo'] == 'A1']

intidorgano strcodorgano  \
29          5295    A13028993   
135         6340    A13037514   

                                        strdescripcion stridadministracion  \
29                                  CANAL DE ISABEL II                   O   
135  FUNDACIÓN DE LA ENERGÍA DE LA COMUNIDAD DE MADRID                   O   

     stragrupa nivel1                                             nivel2  \
29          99  OTROS                                 CANAL DE ISABEL II   
135         99  OTROS  FUNDACIÓN DE LA ENERGÍA DE LA COMUNIDAD DE MADRID   

     nivel3                    descripTrunc  \
29      NaN              CANAL DE ISABEL II   
135     NaN  FUNDACIÓN DE LA ENERGÍA DE ...   

                                        descripTrunc65 tipo_organo  
29                                  CANAL DE ISABEL II          A1  
135  FUNDACIÓN DE LA ENERGÍA DE LA COMUNIDAD DE MADRID          A1

In [15]:
otros_df[otros_df['tipo_organo'] == 'GE']

intidorgano strcodorgano                        strdescripcion  \
33         3313    GE0006082  CENTRO ASOCIADO DE LA UNED EN TUDELA   

   stridadministracion  stragrupa nivel1  \
33                   O         99  OTROS   

                                  nivel2  nivel3  \
33  CENTRO ASOCIADO DE LA UNED EN TUDELA     NaN   

                      descripTrunc                        descripTrunc65  \
33  CENTRO ASOCIADO DE LA UNED ...  CENTRO ASOCIADO DE LA UNED EN TUDELA   

   tipo_organo  
33          GE

Por lo tanto tenemos:

| Código | Tipo de entidad      | Num. de entidades |
|--------|----------------------|-------------------|
| CC     | Cámaras de Comercio  |                71 |
| U0     | Universidades        |                51 |
| GA     | Asociaciones         |                49 |
| I0     | Agencias/consejos    |                29 |
| A1     | Empresas Madrileñas? |                 2 |
| GE     | Centro asociado      |                 1 |

De todo este listado, parece que donde hay más variedad es dentro del grupo I0. En cuanto a las dos empresas madrileñas que aparecen, el Canal de Isabel II y la Fundación de la Energía de la Comunidad de Madrid.


Podemos consultar desde la web las convocatorias lanzadas por el canal usando el código de órgano que acabamos de obtener. La URL de consulta sería la siguiente: https://www.infosubvenciones.es/bdnstrans/GE/es/convocatorias/organo/A13028993

Si queremos por contra obtener quienes han sido los concesionarios, podemos consultar la siguiente URL: https://www.infosubvenciones.es/bdnstrans/GE/es/concesiones/organo/A13028993


# Potencial de mejora
Este tipo de información sólo se ofrece a través del interfaz gráfico de la web y no es posible obtener en formato JSON como el resto de datos. Sería muy interesante poder consultar esta informacía vía API también.